proposals/lip-0048.md

```diff
@@ -25,15 +25,34 @@ This LIP defines the fee system in a modular way, as currently [used in the Lisk
 
 ## Rationale
 
+The fee structure presented here serves to allow defining sufficient cost to manage state growth and prevent transaction pool spamming. In principle, any transaction resulting in a state change should incur a fee. This reasoning is implemented in the `payFee` method, which can be invoked by other modules.
```

Comment:
```suggestion
The fee structure presented here serves to allow defining sufficient cost to manage state growth and prevent transaction pool spamming. In principle, any transaction resulting in a state change could pay a fee. This reasoning is implemented in the `payFee` method, which can be invoked by other modules.
```

Reply:
Agreed. Maybe this?
```suggestion
The fee structure presented here serves to allow defining sufficient cost to manage state growth and prevent transaction pool spamming. In principle, any transaction resulting in a state increase should pay a fee. This reasoning is implemented in the `payFee` method, which can be invoked by other modules.
```

Reply:
Maybe I would avoid mentioning state growth altogether. Transactions should pay a fee for several reasons, e.g. state growth, computational effort from the node, bandwidth, block space, etc. What I would clearly mention is that we do not have an automatic way of calculating the fee of a transaction for each one of these costs. Instead, we only calculate the fee for the block space automatically (similarly to bitcoin if you want) with the min fee, and leave flexibility to developers to assign a fee for the other cases "by hand". For instance, in the Token module we assign by hand an initialization fee which covers for the state growth. In principle, we could also assign a fee each time an expensive calculation is required (e.g. verifying an inclusion proof).

Reply:
I tried to summarize your idea, please check the latest commit.